scripts/cfos/make_dataset_data_tables.py

```diff
@@ -143,57 +108,98 @@ def load_excel_input(excel, expected_dataset_cols):
         else:
             print(e)
         exit()
-    
+
     print("Success: Excel file has been loaded into a dataframe.")
-    return dataset_metadata_df
+    return raw_dataset_df #dataset_metadata_df
 
 
-def get_expected_columns_list(schema_json, dataset_name):
+def parse_config_file(schema_json):
     """Get list of expected columns in excel based on selected dataset schema."""
-
     # read schema into dictionary
-    with open(schema_json) as schema:
-        schema_dict = json.load(schema)
-
-    # get list of tables for a given dataset name
-    dataset_table_names = list(schema_dict[dataset_name].keys())
+    try: 
+        with open(schema_json) as schema:
+            config_dict = json.load(schema)
+    
+    except FileNotFoundError as e:
+        # if the value error is about missing columns in input file
+        exit(e)
+    
 
-    # get all cols (attributes) from all tables for dataset
-    # returns nested list of lists with column names
-    nested_dataset_cols = [schema_dict[dataset_name][table] for table in dataset_table_names]
-    # unnest for flat list of columns to parse from excel file
-    expected_dataset_cols = [col for table_cols in nested_dataset_cols for col in table_cols]
+    schema_dict = config_dict["schema_definitions"]
```

Comment:
why not extract the particular dataset's schema_dict here? rather than having to do `schema_dict[args.dataset_name]` when you feed it into the validate function later on?

Reply:
Great suggestion, I like this better. :)